### PR TITLE
Feat/tt 9265/enterprise portal s3 storage

### PR DESCRIPTION
**Pull request by andrei-tyk**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tyk-single-dc/values.yaml [hunk 1]
```diff
@@ -601,16 +601,30 @@ tyk-enterprise-portal:
...
   storage:
+    # User can set the storage type for portal.
+    # Supported types: fs, s3, db
+    type: "fs"
     database:
       type: "sqlite3"
       connectionString: "db/portal.db"
+    s3:
+      type: "s3"
+      awsAccessKeyid: your-access-key
```
Comment (komalsukhani) [inline tyk-single-dc/values.yaml:619]: Should we support setting of AWS credentials using secret?

File: components/tyk-enterprise-portal/values.yaml [hunk 2]
```diff
@@ -32,12 +32,28 @@ license: ""
...
+kind: StatefulSet
 storage:
+  # User can set the storage type for portal.
+  # Supported types: fs, s3, db
+  type: "fs"
   database:
     type: "sqlite3"
     connectionString: "db/portal.db"
+  s3:
+    type: "s3"
```
Comment (komalsukhani) [inline components/tyk-enterprise-portal/values.yaml:45]: This is not used. Can we remove it?

File: tyk-single-dc/values.yaml [hunk 3]
```diff
@@ -601,16 +601,30 @@ tyk-enterprise-portal:
...
+  kind: StatefulSet
   storage:
+    # User can set the storage type for portal.
+    # Supported types: fs, s3, db
+    type: "fs"
     database:
       type: "sqlite3"
       connectionString: "db/portal.db"
+    s3:
+      type: "s3"
```
Comment (komalsukhani) [inline tyk-single-dc/values.yaml:618]: This is not used. Can we remove it?

File: tyk-single-dc/values.yaml [hunk 4]
```diff
@@ -601,16 +601,30 @@ tyk-enterprise-portal:
...
+  # Enterprise portal can be deployed as StatefulSet or as Deployment
+  kind: StatefulSet
   storage:
+    # User can set the storage type for portal.
+    # Supported types: fs, s3, db
+    type: "fs"
     database:
       type: "sqlite3"
       connectionString: "db/portal.db"
+    s3:
```
Comment (komalsukhani) [inline tyk-single-dc/values.yaml:617]: Can you add a description comment for s3?

File: tyk-single-dc/values.yaml (hunk 1 above)
Comment (andrei-tyk) [reply tyk-single-dc/values.yaml:619]: Added :D

File: tyk-single-dc/values.yaml (hunk 4 above)
Comment (andrei-tyk) [reply tyk-single-dc/values.yaml:639]: Added

**Review by komalsukhani — APPROVED** [review]

File: components/tyk-enterprise-portal/values.yaml [hunk 5]
```diff
@@ -32,12 +32,31 @@ license: ""
...
+  # User can set the storage type for portal.
+  # Supported types: fs, s3, db
+  type: "fs"
   database:
     type: "sqlite3"
     connectionString: "db/portal.db"
+  # Configuration values for using s3 as storage for Tyk Enterprise Portal
+  # In case you want to provide the key ID and access key via secrets please
+  # refer to the existing secret inside the helmchart or the
+  # .Values.global.secrets.useSecretName variable
```
Comment (komalsukhani) [inline components/tyk-enterprise-portal/values.yaml:47]: One last thing. 
Can you define in the comment of `useSecretName` which keys need to be set in the secret?

**Review by buraksekili — APPROVED** [review]

**Pull request merged**